packages/opal-server/opal_server/scopes/api.py

```diff
@@ -0,0 +1,79 @@
+from typing import Optional, cast
+
+from fastapi import APIRouter, HTTPException, Path, Query, Response, status
+from opal_common.schemas.data import DataSourceConfig
+from opal_common.schemas.policy import PolicyBundle
+from opal_common.schemas.policy_source import GitPolicySource
+from opal_common.schemas.scopes import Scope
+from opal_server.config import opal_server_config
+from opal_server.git_fetcher import BadCommitError, GitPolicyFetcher
+from opal_server.scopes.scope_repository import ScopeNotFoundError, ScopeRepository
+
+
+def init_scope_router(scopes: ScopeRepository):
+    router = APIRouter()
+
+    @router.put("", status_code=status.HTTP_201_CREATED)
+    async def put_scope(*, scope_in: Scope):
+        await scopes.put(scope_in)
+
+        from opal_server.worker import sync_scope
+
+        sync_scope.delay(scope_in.scope_id)
+
+        return Response(status_code=status.HTTP_201_CREATED)
```

Comment:
i think you should wait for the celery worker to confirm the creation before returning 201?

you can do this in celery quite easily (simply function return code?)

Reply:
maybe open a ticket to make "clones" or scopes inspectable later? to verify that indeed we cloned ok, etc.